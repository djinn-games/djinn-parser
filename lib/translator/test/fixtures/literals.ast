{
  "type": "Program",
  "name": {
    "type": "Identifier",
    "name": "literals",
    "line": 1
  },
  "body": {
    "type": "Body",
    "sentences": [
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "Literal",
          "line": 3,
          "dataType": "int",
          "value": 2
        },
        "line": 3
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "Literal",
          "line": 4,
          "dataType": "str",
          "value": "hello"
        },
        "line": 4
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "Literal",
          "line": 5,
          "dataType": "float",
          "value": 2.5
        },
        "line": 5
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "Literal",
          "line": 6,
          "dataType": "bool",
          "value": true
        },
        "line": 6
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "Literal",
          "line": 7,
          "dataType": "bool",
          "value": false
        },
        "line": 7
      }
    ],
    "line": 2
  },
  "line": 1
}