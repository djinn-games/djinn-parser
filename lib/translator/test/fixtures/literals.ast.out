{
  "type": "Program",
  "body": [
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "Literal",
        "value": 2,
        "raw": "2"
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "Literal",
        "value": "hello",
        "raw": "\"hello\""
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "Literal",
        "value": 2.5,
        "raw": "2.5"
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "Literal",
        "value": true,
        "raw": "true"
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "Literal",
        "value": false,
        "raw": "false"
      }
    }
  ]
}
